Dockerfile

```diff
@@ -0,0 +1,53 @@
+FROM postgres_base AS base
+
+###
+### BUILDER
+###
+FROM base AS builder
+ARG POSTGRES_VERSION
+
+RUN apt-get update -qq && \
+  apt-get install -y \
+    postgresql-server-dev-${POSTGRES_VERSION} \
+    build-essential libreadline-dev zlib1g-dev flex bison libxml2-dev libxslt-dev \
+    libssl-dev libxml2-utils xsltproc pkg-config libc++-dev libc++abi-dev libglib2.0-dev \
+    libtinfo5 cmake libstdc++-12-dev liblz4-dev ccache ninja-build && \
+  rm -rf /var/lib/apt/lists/*
+
+WORKDIR /build
+
+ENV PATH=/usr/lib/ccache:$PATH
+ENV CCACHE_DIR=/ccache
+
+# A more selective copy might be nice, but the git submodules are not cooperative.
+# Instead, use .dockerignore to not copy files here.
+COPY . .
```

Comment:
I think it would be good to automatically run `CREATE EXTENSION pg_duckdb` and configure `shared_preload_libraries=pg_duckdb`. That'll make it easier to run.

Using an initialization script should make automating that possible I think: https://github.com/docker-library/docs/blob/master/postgres/README.md#initialization-scripts

Reply:
yeah we can handle that at the docker compose level